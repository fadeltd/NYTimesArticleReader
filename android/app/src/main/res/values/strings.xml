<resources>
    <string name="app_name">NYTimesArticleReader</string>
</resources>
