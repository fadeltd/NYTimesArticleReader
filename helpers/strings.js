/**
 * Helper class for strings manipulation
 */
export default class Strings {
  static toTitleCase(text) {
    return text.replace(/\w\S*/g, (word) => {
      return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
    });
  }
  static getParams(url, params){
    let query = '';
    for(let key in params){
      query += `${key}=${params[key]}&`;
    }
    return `${url}?${query}`;
  }
}
