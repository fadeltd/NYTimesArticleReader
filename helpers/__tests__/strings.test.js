import strings from '../strings';

describe('To Title Case', () => {
  it('should return the string of By The Associated Press)', () => {
    expect(strings.toTitleCase('By THE ASSOCIATED PRESS')).toMatchSnapshot();
  });

  it('should return the string of By Reuters)', () => {
    expect(strings.toTitleCase('By REUTERS')).toMatchSnapshot();
  });

  it('should return null', () => {
    expect(strings.toTitleCase('')).toMatchSnapshot();
  });
});

describe('Get Params', () => {
  it('should return the string of https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=newest&q=Brazil&', () => {
    expect(strings.getParams('https://api.nytimes.com/svc/search/v2/articlesearch.json', {
      'api-key': '3304a9018a9a4cd8a6c587cd2060e092',
      'page': 0,
      'sort': 'newest',
      'q': 'Brazil'
    })).toMatchSnapshot();
  });

  it('should return the string of https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=oldest&', () => {
    expect(strings.getParams('https://api.nytimes.com/svc/search/v2/articlesearch.json', {
      'api-key': '3304a9018a9a4cd8a6c587cd2060e092',
      'page': 0,
      'sort': 'oldest',
    })).toMatchSnapshot();
  });

  it('should return null', () => {
    expect(strings.getParams('', {})).toMatchSnapshot();
  });
});