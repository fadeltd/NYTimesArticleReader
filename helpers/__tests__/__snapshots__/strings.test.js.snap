// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Params should return null 1`] = `"?"`;

exports[`Get Params should return the string of https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=newest&q=Brazil& 1`] = `"https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=newest&q=Brazil&"`;

exports[`Get Params should return the string of https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=oldest& 1`] = `"https://api.nytimes.com/svc/search/v2/articlesearch.json?api-key=3304a9018a9a4cd8a6c587cd2060e092&page=0&sort=oldest&"`;

exports[`To Title Case should return null 1`] = `""`;

exports[`To Title Case should return the string of By Reuters) 1`] = `"By Reuters"`;

exports[`To Title Case should return the string of By The Associated Press) 1`] = `"By The Associated Press"`;
